### PR TITLE
Implement ecHj to list highlight rules in json format

### LINKED ISSUE #15996: Implement ecHj to list highlight rules in json format
maybe useful for cutter and the webuis

Also, i would like to drop this uppercase 'H' from the naming, and it should be probably  moved to CH because it is stored as an RAnalMeta.

Comments on this @thestr4ng3r  ?

### PR DESCRIPTION
**Pull request by c-urly**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/radareorg/radare2/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/radareorg/radare2/blob/master/DEVELOPERS.md#code-style)
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [radare2 book](https://github.com/radareorg/radare2book) with the relevant information (if needed)

**Detailed description**

- Implement ecHj command to list highlight rules in json format.

- Usage:
Screenshot 2020-03-03 at 9 39 29 PM

...

**Test plan**

...

**Closing issues**
closes #15996 

...

### CODE REVIEW

File: libr/anal/meta.c [hunk 1]
```diff
@@ -653,6 +659,7 @@ R_API void r_meta_print(RAnal *a, RAnalMetaItem *d, int rad, PJ *pj, bool show_f
 			}
 
 			pj_end (pj);
+			free(rgb_str);
```
Comment (radare) [inline libr/anal/meta.c]: missing space before (

File: libr/anal/meta.c [hunk 2]
```diff
@@ -626,12 +626,18 @@ R_API void r_meta_print(RAnal *a, RAnalMetaItem *d, int rad, PJ *pj, bool show_f
...
-			pj_k (pj, "name");
-			if (d->type == 's' && (base64_str = r_base64_encode_dyn (d->str, -1))) {
+			pj_k (pj, "color");
+			ut8 r = 0, g = 0, b = 0, A = 0;
+			char* rgb_str;
+			const char *esc = strchr (d->str, '\x1b');
+			r_cons_rgb_parse (esc, &r, &g, &b, &A);
+			rgb_str = r_cons_rgb_tostring(r, g, b);
+
+			if (d->type == 's' && (base64_str = r_base64_encode_dyn (rgb_str, -1))) {
```
Comment (radare) [inline libr/anal/meta.c]: do not assign inside an if statement

File: libr/anal/meta.c [hunk 3]
```diff
@@ -626,12 +626,18 @@ R_API void r_meta_print(RAnal *a, RAnalMetaItem *d, int rad, PJ *pj, bool show_f
...
 			pj_ks (pj, "type", r_meta_type_to_string (d->type));
 
-			pj_k (pj, "name");
-			if (d->type == 's' && (base64_str = r_base64_encode_dyn (d->str, -1))) {
+			pj_k (pj, "color");
+			ut8 r = 0, g = 0, b = 0, A = 0;
+			char* rgb_str;
+			const char *esc = strchr (d->str, '\x1b');
+			r_cons_rgb_parse (esc, &r, &g, &b, &A);
+			rgb_str = r_cons_rgb_tostring(r, g, b);
```
Comment (radare) [inline libr/anal/meta.c]: missing space before (

Comment (radare) [inline libr/anal/meta.c]: define and assign in the same line char *rgb_str ...

File: libr/anal/meta.c [hunk 4]
```diff
@@ -626,12 +626,17 @@ R_API void r_meta_print(RAnal *a, RAnalMetaItem *d, int rad, PJ *pj, bool show_f
 			pj_kn (pj, "offset", d->from);
 			pj_ks (pj, "type", r_meta_type_to_string (d->type));
 
-			pj_k (pj, "name");
-			if (d->type == 's' && (base64_str = r_base64_encode_dyn (d->str, -1))) {
+			pj_k (pj, "color");
```
Comment (radare) [inline libr/anal/meta.c]: do only this on highlight metas, otherwise you are breaking comments and many other metadatas. see the XX] tests in travis logs

**Comment by radare** [comment]
fixed in https://github.com/radareorg/radare2/pull/16123

**Comment by radare** [comment]
done!! thx

File: libr/anal/meta.c (hunk 4 above)
Comment (c-urly) [reply libr/anal/meta.c]: Ah..yes i had that question but you didn't mention in review so i thought we have to change it to color.

**Pull request closed without merging**